iothub/device/src/Transport/AmqpIot/AmqpIotLinkFactory.cs

```diff
@@ -9,17 +9,9 @@ namespace Microsoft.Azure.Devices.Client.Transport.AmqpIot
 {
     internal class AmqpIotLinkFactory : ILinkFactory
     {
-        private static readonly AmqpIotLinkFactory s_instance = new AmqpIotLinkFactory();
+        private static readonly Lazy<AmqpIotLinkFactory> s_instance = new(() => new AmqpIotLinkFactory());
```

Comment:
Rather that rely on not referencing the type before the static ctor gets called, we can use the defined pattern for delayed construction.